src/controllers/mainController.ts

```diff
@@ -202,6 +199,7 @@ export default class MainController implements vscode.Disposable {
                         editor.edit(editBuilder => {
                             editBuilder.replace(editor.selection, selectStatement);
                         }).then(() => this.onRunQuery());
+                        return this.connectionManager.connectionStore.removeRecentlyUsed(<IConnectionProfile>connectionCreds);
                     }
```

Comment:
So you're removing it because you don't want it to show up in the MRU? Why not just add a flag to the call to connect that tells it not to save it in the first place? That seems a lot cleaner. Right now you might even have the weird situation where it appears briefly in the MRU list until this call here is actually made.

Reply:
I don't want to change the existing `User Settings` and `MRu` logic as it can be prone to many side effects. Also, we will never have that issue because the object explorer updates the connections when refreshed.

Reply:
It seems like a "code smell" to have code that does something then has to call another function to partially undo what was just previously done.  If we have code we're afraid to change due to potential side-effects that seems like a problem.  Presumably some refactoring and additional tests would be a prudent mitigation?